app/src/main/java/com/example/pokemon/ui/DetailActivity.kt

```diff
@@ -1,36 +1,39 @@
 package com.example.pokemon.ui
 
+import android.annotation.SuppressLint
 import android.content.res.ColorStateList
 import android.os.Bundle
 import android.view.View
 import android.widget.ImageView
 import android.widget.LinearLayout
 import android.widget.Toast
-import androidx.appcompat.app.AppCompatActivity
 import com.bumptech.glide.Glide
 import com.example.pokemon.R
-import com.example.pokemon.contract.PokemonDetailContract
+import com.example.pokemon.contract.DetailContract
 import com.example.pokemon.model.Pokemon
-import com.example.pokemon.presenter.PokemonDetailPresenterImpl
 import com.google.android.material.button.MaterialButton
 import com.google.gson.Gson
 import com.varunest.sparkbutton.SparkEventListener
+import dagger.android.support.DaggerAppCompatActivity
 import kotlinx.android.synthetic.main.activity_pokemon_detail.*
+import javax.inject.Inject
 
-class PokemonDetailActivity : AppCompatActivity(), PokemonDetailContract.PokemonDetailView {
+class DetailActivity : DaggerAppCompatActivity(),
+    DetailContract.View {
 
-    private var pokemonDetailPresenterImpl: PokemonDetailPresenterImpl? = null
+    @Inject
+    lateinit var detailPresenter: DetailContract.Presenter
+    private var pokemon: Pokemon? = null
```

Comment:
why do we need this variable ?

Reply:
pokemon? For taking the pokemon which is coming from the intent. If the details were not loaded and the user clicked the favorite button, it wouldn't reflect in db as it will get updated again when details load, so I took a global var and stored the pokemon and I'm checking if the height attribute is null i.e, the api isn't loaded yet, so I'm making the button disabled.